docs/desktop/wpf/fundamentals/xaml.md

```diff
@@ -1,11 +1,348 @@
+<!--TODO For details, see [White space processing in XAML](../../../framework/xaml-services/whitespace-processing-in-xaml.md).-->
+
+## Markup extensions
+
+Markup extensions are a XAML language concept. When used to provide the value of an attribute syntax, curly braces (`{` and `}`) indicate a markup extension usage. This usage directs the XAML processing to escape from the general treatment of attribute values as either a literal string or a string-convertible value.
+
+<!-- TODO LINK CLEAN UP -->
+
+The most common markup extensions used in WPF application programming are [Binding](binding-markup-extension.md), used for data binding expressions, and the resource references [StaticResource](staticresource-markup-extension.md) and [DynamicResource](dynamicresource-markup-extension.md). By using markup extensions, you can use attribute syntax to provide values for properties even if that property does not support an attribute syntax in general. Markup extensions often use intermediate expression types to enable features such as deferring values or referencing other objects that are only present at run-time.
+
+For example, the following markup sets the value of the <xref:System.Windows.FrameworkElement.Style%2A> property using attribute syntax. The <xref:System.Windows.FrameworkElement.Style%2A> property takes an instance of the <xref:System.Windows.Style> class, which by default could not be instantiated by an attribute syntax string. But in this case, the attribute references a particular markup extension, [StaticResource](staticresource-markup-extension.md). When that markup extension is processed, it returns a reference to a style that was previously instantiated as a keyed resource in a resource dictionary.
+
+[!code-xaml[FEResourceSH_snip#XAMLOvwShortResources](~/samples/snippets/csharp/VS_Snippets_Wpf/FEResourceSH_snip/CS/page1.xaml#xamlovwshortresources)]
+[!code-xaml[FEResourceSH_snip#XAMLOvwShortResources2](~/samples/snippets/csharp/VS_Snippets_Wpf/FEResourceSH_snip/CS/page1.xaml#xamlovwshortresources2)]
+[!code-xaml[FEResourceSH_snip#XAMLOvwShortResources3](~/samples/snippets/csharp/VS_Snippets_Wpf/FEResourceSH_snip/CS/page1.xaml#xamlovwshortresources3)]
+
+For a reference listing of all markup extensions for XAML implemented specifically in WPF, see [WPF XAML Extensions](wpf-xaml-extensions.md). For a reference listing of the markup extensions that are defined by System.Xaml and are more widely available for .NET Framework XAML implementations, see [XAML Namespace (x:) Language Features](../../../framework/xaml-services/xaml-namespace-x-language-features.md). <!-- TODO For more information about markup extension concepts, see [Markup Extensions and WPF XAML](markup-extensions-and-wpf-xaml.md). -->
```

Comment:
should this say ".NET Core Xaml implementations" or maybe just ".NET Xaml implementations"?